src/api_manager/service_control/aggregated.cc

```diff
@@ -377,21 +377,10 @@ void Aggregated::Check(
     if (status.ok()) {
       Status status = Proto::ConvertCheckResponse(
           *response, service_control_proto_.service_name(), &response_info);
-      // If server replied with either invalid api_key or not activated service,
-      // the request is rejected even allow_unregistered_calls is true. Most
-      // likely, users provide a wrong api key. By failing the request, the
-      // users will be notified with the error and have chance to correct it.
-      // Otherwise, the Report call will fail. It is very hard to notice and
-      // debug the Report failure.
-      if (allow_unregistered_calls && response_info.is_api_key_valid &&
-          response_info.service_is_activated) {
-        on_done(Status::OK, response_info);
-      } else {
-        on_done(status, response_info);
-      }
+      on_done(status, response_info);
     } else {
-      // If allow_unregistered_calls is true, it is always OK to proceed.
-      if (allow_unregistered_calls) {
+      // If network_fail_open is true, it is always OK to proceed.
```

Comment:
Does this mean:  the api produce need to config in server_config.proto, instead of service config, for this purpose?

Reply:
Next pr will add a start_esp flag to set it to server config.